src/Compilers/Core/Portable/MetadataReader/MetadataDecoder.cs

```diff
@@ -1476,6 +1476,41 @@ private void DecodeCustomAttributeFieldOrPropTypeOrThrow(ref BlobReader argReade
             throw new UnsupportedSignatureContent();
         }
 
+        /// <exception cref="UnsupportedSignatureContent">If the encoded attribute argument is invalid.</exception>
+        /// <exception cref="BadImageFormatException">An exception from metadata reader.</exception>
+        private TypedConstant DecodeCustomAttributeFixedArgumentOrThrow(ITypeSymbolInternal type, ref BlobReader argReader)
+        {
+            // arrays are allowed only on top-level:
+            if (type.GetElementTypeIfSZArray() is { } elementType)
+            {
+                return DecodeCustomAttributeElementArrayOrThrow(ref argReader, getTypeCode(elementType), (TypeSymbol)elementType, (TypeSymbol)type);
+            }
+
+            return DecodeCustomAttributeElementOrThrow(ref argReader, getTypeCode(type), (TypeSymbol)type);
+
+            SerializationTypeCode getTypeCode(ITypeSymbolInternal type)
+            {
+                if (ReferenceEquals(type, SystemTypeSymbol))
```

Comment:
>ReferenceEquals(type, SystemTypeSymbol) [](http://example.com/codeflow?start=20&length=39)

It looks like `DecodeCustomAttributeParameterTypeOrThrow` uses regular comparison. Consider using it here as well.

Reply:
It uses comparison with `(object)` cast, I guess to ensure it's a reference equality - isn't `ReferenceEquals` call better way to express that?

Reply:
Anyway, I don't mind changing this, just not sure if you think I should use `type == SystemTypeSymbol` or `(object)type == SystemTypeSymbol` (both seem slightly worse to me than `ReferenceEquals` if our intention is to perform reference equality).